ansible/roles/jenkins/tasks/main.yml

```
@@ -0,0 +1,46 @@
+---
+# tasks file for jenkins
+
+- name: Install tools
+  ansible.builtin.import_tasks:
+    file: tools.yml
+
+- name: Install Jenkins
+  ansible.builtin.apt:
+    name: "{{ java_version }}"
+    state: present
+
+- name: Add jenkins apt key
+  ansible.builtin.apt_key:
+    url: "{{ jenkins.jenkins_key }}"
+    state: present
+
+- name: Add jenkins apt repository
+  ansible.builtin.apt_repository:
+    repo: "{{ jenkins.jenkins_repo }}"
+    state: present
+    become: true
+
+- name: Update apt cache
+  ansible.builtin.apt:
+    update_cache: yes
+
+- name: Install Jenkins
+  ansible.builtin.apt:
+    name: jenkins
+    state: present
+
+- name: Start Jenkins
+  ansible.builtin.service:
+    name: jenkins
+    state: started
+
+- name: Wait for jenkins to start
+  wait_for: timeout=60
+  delegate_to: localhost
+
+- name: Get jenkins initial admin password
+  ansible.builtin.shell: "cat /var/lib/jenkins/secrets/initialAdminPassword"
```

command-instead-of-shell: Use shell only when shell functionality is required
```suggestion
  ansible.builtin.command: "cat /var/lib/jenkins/secrets/initialAdminPassword"
```